test/integration/logs.test.js

```diff
@@ -0,0 +1,221 @@
+import chai from "chai";
+import chaiHttp from "chai-http";
+import config from "config";
+import app from "../../server";
+import cleanDb from "../utils/cleanDb";
+import authService from "../../services/authService";
+import userDataFixture from "../fixtures/user/user";
+import addUser from "../utils/addUser";
+import { createOooRequests } from "../fixtures/oooRequest/oooRequest";
+import { createRequest } from "../../models/requests";
+import logsQuery, { addLog } from "../../models/logs";
+import { LOG_ACTION, REQUEST_LOG_TYPE } from "../../constants/requests";
+import { requestsLogs } from "../fixtures/logs/requests";
+import { extensionRequestLogs } from "../fixtures/logs/extensionRequests";
+const { expect } = chai;
+const cookieName = config.get("userToken.cookieName");
+
+const userData = userDataFixture();
+chai.use(chaiHttp);
+
+let authToken;
+let superUserToken;
+let oooRequestData;
+describe("/logs", function () {
+  beforeEach(async function () {
+    const userIdPromises = [addUser(userData[16]), addUser(userData[4])];
+    const [userId, superUserId] = await Promise.all(userIdPromises);
+
+    oooRequestData = { ...createOooRequests, requestedBy: userId };
+    const requestResult = await createRequest(oooRequestData);
+    const requestLog = {
+      type: REQUEST_LOG_TYPE.REQUEST_CREATED,
+      meta: {
+        requestId: requestResult.id,
+        action: LOG_ACTION.CREATE,
+        createdBy: userId,
+        createdAt: Date.now(),
+      },
+      body: requestResult,
+    };
+    await addLog(requestLog.type, requestLog.meta, requestLog.body);
+    await addLogs();
+    authToken = authService.generateAuthToken({ userId });
+    superUserToken = authService.generateAuthToken({ userId: superUserId });
+  });
+
+  afterEach(async function () {
+    await cleanDb();
+  });
+
+  describe("GET /logs", function () {
+    it("should return logs of specific type", function (done) {
+      chai
+        .request(app)
+        .get("/logs/REQUEST_CREATED")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .end(function (err, res) {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(200);
+          expect(res.body.message).to.equal("Logs fetched successfully");
+          expect(res.body.logs).to.have.lengthOf(3);
+          const log = res.body.logs[0];
+          expect(log).to.have.property("meta");
+          expect(log).to.have.property("body");
+          expect(log.type).to.equal("REQUEST_CREATED");
+          return done();
+        });
+    });
+
+    it("should return 401 if user is not logged in", function (done) {
+      chai
+        .request(app)
+        .get("/logs?dev=true")
```

Comment:
- why are we checking this under a feature flag ?

Reply:
I took reference from other routes and were also told every feature has to go under feature flag. 
Do you want me to remove this check?